geo/src/algorithm/stitch.rs

```diff
@@ -0,0 +1,588 @@
+    ///   input set. If you want to do a union on overlapping triangles then c.f. [`SpadeBoolops`].
+    /// - Input triangles should be valid polygons. For a definition of validity
+    ///   c.f. <https://www.postgis.net/workshops/postgis-intro/validity.html>
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// // commented this out since the trait is private and the doc test will fail
+    ///
+    /// //use geo::StitchTriangles;
+    /// //use geo::{Coord, Triangle, polygon};
+    ///
+    /// //let tri1 = Triangle::from([
+    /// //    Coord { x: 0.0, y: 0.0 },
+    /// //    Coord { x: 1.0, y: 0.0 },
+    /// //    Coord { x: 0.0, y: 1.0 },
+    /// //]);
+    /// //let tri2 = Triangle::from([
+    /// //    Coord { x: 1.0, y: 1.0 },
+    /// //    Coord { x: 1.0, y: 0.0 },
+    /// //    Coord { x: 0.0, y: 1.0 },
+    /// //]);
+    ///
+    /// //let result = vec![tri1, tri2].stitch_triangulation();
+    ///
+    /// //assert!(result.is_ok());
+    ///
+    /// //let mp = result.unwrap();
+    ///
+    /// //assert_eq!(mp.0.len(), 1);
+    ///
+    /// //let poly = mp.0[0].clone();
+    /// //// 4 coords + 1 duplicate for closed-ness
+    /// //assert_eq!(poly.exterior().0.len(), 4 + 1);
+    ///
+    /// //let expected = polygon![
+    /// //    Coord { x: 1.0, y: 1.0 },
+    /// //    Coord { x: 0.0, y: 1.0 },
+    /// //    Coord { x: 0.0, y: 0.0 },
+    /// //    Coord { x: 1.0, y: 0.0 },
+    /// //];
+    ///
+    /// //assert_eq!(poly, expected);
+    /// ```
+    ///
+    /// # Additional Notes
+    ///
+    /// If you want to do something more general like a
+    /// [`Boolean Operation Union`](https://en.wikipedia.org/wiki/Boolean_operations_on_polygons)
+    /// you should use the trait `BooleanOps` or `SpadeBoolops`.
+    fn stitch_triangulation(&self) -> TriangleStitchingResult<MultiPolygon<T>>;
+}
+
+impl<S, T> StitchTriangles<T> for S
+where
+    S: AsRef<[Triangle<T>]>,
+    T: GeoFloat,
+{
+    fn stitch_triangulation(&self) -> TriangleStitchingResult<MultiPolygon<T>> {
+        stitch_triangles(self.as_ref().iter())
+    }
+}
+
+// main stitching algorithm
+fn stitch_triangles<'a, T, S>(triangles: S) -> TriangleStitchingResult<MultiPolygon<T>>
+where
+    T: GeoFloat + 'a,
+    S: Iterator<Item = &'a Triangle<T>>,
+{
+    let lines = triangles.flat_map(ccw_lines).collect::<Vec<_>>();
+
+    let boundary_lines = find_boundary_lines(lines);
+    let stitched_multipolygon = stitch_multipolygon_from_lines(boundary_lines)?;
+
+    let polys = stitched_multipolygon
+        .into_iter()
+        .map(find_and_fix_holes_in_exterior)
+        .collect::<Vec<_>>();
+
+    Ok(MultiPolygon::new(polys))
+}
+
+/// special cased algorithm for finding the winding of a triangle
+///
+/// note that we assume the triangle is valid, i.e. it's not three points on a flat line
+fn triangle_winding_order<T: GeoFloat>(tri: &Triangle<T>) -> WindingOrder {
```

Comment:
It would be cool to put it somewhere else but the `Winding` trait is currently pretty much tied to `LineString`s so that I can't really add it. Creating a `LineString` from the Triangle is probably also a no-no, since we want to avoid the allocation. Doing any refactoring to make this possible is probably also overkill for the scope of this PR. 

I could move the function into the module though to make it more discoverable and also prevent anyone else from rewriting it in the module. I'm going for that 👍🏼